packages/browser-wallet/src/popup/pages/Account/TransactionLog/TransactionList.tsx

```diff
@@ -140,6 +143,44 @@ function InfiniteTransactionList({
     );
 }
 
+/**
+ * Update an existing list of descending transactions with a new list of ascending transactions,
+ * ensuring that any overlaps are updated with the values from the new list of transactions.
+ */
+function updateTransactionsWithNewTransactions(
+    existingTransactions: BrowserWalletTransaction[],
+    newTransactions: BrowserWalletTransaction[]
+) {
+    const existingHashes = existingTransactions.map((trs) => trs.transactionHash);
```

Comment:
Please filter out rewards/undefined hashes here so that the rewards don't update each other.